### PR TITLE
Let getprovisionsourcestorage read state

### PR DESCRIPTION
**Pull request by yaacov**
getProvisionSourceStorage method will need to fetch golden images and common templates from current state

Part of: #5977

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts [hunk 1]
```diff
@@ -94,11 +96,16 @@ const getUrlStorage = (storageClassConfigMap: ConfigMapKind) => {
   };
 };
 
-export const getProvisionSourceStorage = (
-  provisionSource: ProvisionSource,
-  iStorageClassConfigMap: any,
-): VMWizardStorage => {
+export const getProvisionSourceStorage = (state: any, id: string): VMWizardStorage => {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts:99]: we already have a similar method which is doing something different (`iGetProvisionSourceStorage`). Can we rename this to `getNewProvisionSourceStorage` ?

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/prefill-vm-template-state-update.ts [hunk 2]
```diff
@@ -249,10 +249,7 @@ export const prefillVmTemplateUpdater = ({ id, dispatch, getState }: UpdateOptio
     });
     storagesUpdate.unshift(...templateStorages);
   } else {
-    const newSourceStorage = getProvisionSourceStorage(
-      iGetProvisionSource(state, id),
-      iGetLoadedCommonData(state, id, VMWizardProps.storageClassConfigMap),
-    );
+    const newSourceStorage = getProvisionSourceStorage(state, id);
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/prefill-vm-template-state-update.ts:252]: +1 this simplifies it nicely

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts]: sure :+1: done

**Pull request merged**